mlflow/langchain/__init__.py

```diff
@@ -661,18 +661,21 @@ def _convert_ndarray_to_list(data):
             return data
 
         if isinstance(data, pd.DataFrame):
-            return _convert_ndarray_to_list(data.to_dict(orient="records"))
+            data = data.to_dict(orient="records")
 
         data = _convert_ndarray_to_list(data)
-        if isinstance(self.lc_model, lc_runnables_types()):
-            return [data]
-        if isinstance(data, list) and (
-            all(isinstance(d, str) for d in data) or all(isinstance(d, dict) for d in data)
-        ):
-            return data
+        if not isinstance(data, list):
+            # if the input data is not a list (i.e. single input),
+            # we still need to convert it to a one-element list `[data]`
+            # because `process_api_requests` only accepts list as valid input.
+            # and in this case,
+            # we should return the first element of the inference result
+            # because we change input `data` to `[data]`
+            return [data], True
+        if isinstance(data, list):
```

Comment:
Fix:

We don't need the checking `all(isinstance(d, str) for d in data) or all(isinstance(d, dict) for d in data)` for elements in the list, 

because langchain model can accept any types other than dict / str, e.g. `RunnableLambda` model.

Reply:
The basic types like Chain and retriever does have assumption on the input types 🤔 While it should raise error if the input type doesn't match when sending to the model, we should be fine here!

Reply:
> The basic types like Chain and retriever does have assumption on the input types

`_enforce_schema` does input schema validation if `input_example` data is logged when logging model. Adding additional input type checking here is redundant and fragile, langchain has many kinds of models and they accepts various types input.

Reply:
@WeichenXu123 Do we have any test cases for something like `RunnableLambda` that doesn't accept dict / str? Can we add those test cases? Not seeing them in the PR

Reply:
We don't need to add new test, after Fix-1 is applied, this bug is captured by test test_save_load_runnable_lambda